aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -77,6 +77,38 @@ module aptos_std::simple_map {
         vector::push_back(&mut map.data, Element { key, value });
     }
 
+    public inline fun upsert<Key: store, Value: store>(
```

Comment:
What is the tradeoff between this vs having a non-inline function that returns the old values (as Option<Key> and Option<Value>)?